applications/MIT.cc

```diff
@@ -201,31 +199,53 @@ flag_local_refinement(true)
   cfl_output_file << "Time" << "," << "CFL" << std::endl;
 }
 
-template <int dim>
-void MITBenchmark<dim>::make_grid(const unsigned int n_global_refinements)
+template <>
+void MITBenchmark<2>::make_grid()
 {
   TimerOutput::Scope  t(*this->computing_timer, "Problem: Setup - Triangulation");
 
-  // Reads the structured mesh done in gmsh. This mesh is modelled after
-  // the one named "1R" by the benchmark's authors.
-  GridIn<dim> grid_in;
-  grid_in.attach_triangulation(this->triangulation);
-  std::ifstream triangulation_file("MIT.msh");
-  grid_in.read_msh(triangulation_file);
+  constexpr int dim = 2;
+  constexpr double tol = 1e-12;
+
+  GridGenerator::subdivided_hyper_rectangle(this->triangulation,
+                                            {1u, 8u},
+                                            Point<dim>(0., 0.),
+                                            Point<dim>(1., 8.));
+
+  for (auto &face: triangulation.active_face_iterators())
+    if (face->at_boundary())
+    {
+      const Point<dim> center = face->center();
+
+      if (std::abs(center[1]) > tol && std::abs(center[1] - 8.0) > tol)
+      {
+        if (std::abs(center[0]) < tol)
+          face->set_boundary_id(left_bndry_id);
+        else if (std::abs(center[0] - 1.0) < tol)
+          face->set_boundary_id(right_bndry_id);
+        else
+          Assert(false, ExcInternalError());
+      }
+      else if (std::abs(center[1]) < tol)
+        face->set_boundary_id(bottom_bndry_id);
+      else if (std::abs(center[1] - 8.0) < tol)
+        face->set_boundary_id(top_bndry_id);
+      else
+        Assert(false, ExcInternalError());
+    }
```

Comment:
I know this mesh is a quick fix but I am not really satisfied with it. Its improvement in order to reflect the tailored structured mesh should be added to the to-do list.

Reply:
The mesh was optimized in a3b6a67. Let me know your opinion on the current version.